tools/scripts/ManifestBuilder.rb

```diff
@@ -0,0 +1,436 @@
+require 'aws-sdk-s3'
+require 'ruby-progressbar'
+require 'optparse'
+require 'parallel'
+require_relative '../../deployment'
+require_relative '../../lib/cdo/cdo_cli'
+require_relative '../../lib/cdo/png_utils'
+require_relative './constants'
+include CdoCli
+
+DEFAULT_S3_BUCKET = 'cdo-animation-library'.freeze
+DEFAULT_OUTPUT_FILE = "#{`git rev-parse --show-toplevel`.strip}/apps/src/p5lab/gamelab/animationLibrary.json".freeze
+SPRITELAB_OUTPUT_FILE = "#{`git rev-parse --show-toplevel`.strip}/apps/src/p5lab/spritelab/spriteCostumeLibrary.json".freeze
+DOWNLOAD_DESTINATION = '~/cdo-animation-library'.freeze
+SPRITE_COSTUME_LIST = SPRITE_LAB_ANIMATION_LIST
+
+class Hash
+  # Like Enumerable::map but returns a Hash instead of an Array
+  def hmap(&block)
+    Hash[map {|k, v| yield k, v}]
+  end
+
+  # Drop a key from the hash, returning the hash (destructive)
+  def omit!(key)
+    tap {|hs| hs.delete(key)}
+  end
+end
```

Comment:
Wow, I didn't know we were modifying the built-in Hash class like this. We should probably do some follow-up work to clean this class up a bit

Reply:
haha yeah this file needs some love and I'm not doing it any favors in this project. I'd like to take a couple hours after I'm done to clean it up a bit.